docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/kubernetes.md

```diff
@@ -113,14 +138,315 @@ helm --namespace dagster-cloud upgrade agent \
+  locations:
+    - location_name: project-a
+      ...
+      agent_queue: on-prem-agent-queue
+    - location_name: project-b
+      ...
+      agent_queue: cloud-agent-queue
+  ```
+
+:::tip
+Code locations without an `agent_queue` will be routed to a default queue. You pick an agent to serve this default queue using the `includeDefaultQueue` setting:
+
+```yaml file=values.yaml
+dagsterCloud:
+  agentQueues:
+    includeDefaultQueue: true
+```
+:::
+
+</TabItem>
+<TabItem value="separate-agents" label="Separate deployments per cluster">
+
+If you want completely separate environments with their own asset graph, run history, and access controls you should create different Dagster+ deployments. Separate deployments are common for isolated tenants or separate dev, stage, and prod environments. To create separate deployments:
+
+- Navigate to the deployments page for your organization: https://your-organizaiton.dagster.plus/org-settings/deployments
+
+- Click "New Deployment"
+
+- Follow the preceding instructions for creating and deploying agent for this deployment
+
+</TabItem>
+</Tabs>
+
+### Request resources such as CPU, memory, or GPUs
+
+:::note
+This guide shows how to request more resources for the Dagster run of an asset. If your Dagster asset uses the [Kubernetes pipes client](/todo) you should instead request resources like [this](https://github.com/dagster-io/hooli-data-eng-pipelines/blob/master/hooli_data_eng/assets/forecasting/__init__.py#L257-L288).
+:::
+
+:::tip
+Dagster+ makes it easy to monitor CPU and memory used by code location servers and individual runs. Follow [this guide](/todo) for details.
+:::
+
+First determine if you want to change the requested resource for everything in a code location, or for a specific job or asset.
+
+<Tabs>
+
+<TabItem value="code-location-resource" label="Resources for everything in a code location">
+
+Modify the [`dagster_cloud.yaml` file](/todo) in your project's Git repository:
+
+```yaml file=dagster_cloud.yaml
+locations:
+  - location_name: cloud-examples
+    image: dagster/dagster-cloud-examples:latest
+    code_source:
+      package_name: dagster_cloud_examples
+    container_context:
+      k8s:
+        server_k8s_config:
+          container_config:
+            resources:
+              limits:
+                cpu: 500m
+                memory: 2560Mi
+        run_k8s_config:
+          container_config:
+            resources:
+              limits:
+                cpu: 500m
+                memory: 2560Mi
+                nvidia.com/gpu: 1
+```
+
+The `server_k8s_config` section sets resources for the code location servers, which is where schedule and sensor evaluations occur.
+
+The `runs_k8s_config` section sets resources for the individual runs.
+
+Requests are used by Kubernetes to determine which node to place a pod on, and limits are a strict upper bound on how many resources a pod can use while running. We recommend using limits in most cases.
```

Comment:
both are usually a good idea in my experience - requests keep things from getting too packed on one pod